Makefile

```diff
@@ -424,8 +430,12 @@ tests/%.prove-legacy: tests/%
 	    --no-provex --format-failures $(KPROVE_OPTS) --concrete-rules-file $(dir $@)concrete-rules.txt
 
 tests/specs/examples/erc20-spec/haskell/erc20-spec-kompiled/timestamp: tests/specs/examples/erc20-bin-runtime.k
-tests/specs/examples/erc20-bin-runtime.k: tests/specs/examples/ERC20.sol $(KEVM_LIB)/$(haskell_kompiled) $(KEVM_LIB)/kevm-pyk.py
-	$(KEVM) solc-to-k $< ERC20 > $@
+tests/specs/examples/erc20-bin-runtime.k: tests/specs/examples/ERC20.sol $(KEVM_LIB)/$(haskell_kompiled) build-kevm-pyk
```

Comment:
The reason to have `$(KEVM_LIB)/kevm-pyk.py` instead of `build-kevm-pyk` is because on rebuilds (run this target twice), it will re-trigger a build of `bulid-kevm-pyk` because it's not a file, but `$(KEVM_LIB)/kevm-pyk.py` is a file that it can check the timestamp of.

At least, that's what I am remembering now, but it may be incorrect because `build-kevm-pyk` is marked as `.PHONY`.

You should double-check that on re-builds it doesn't end up _always_ rebuilding the `build-kevm-pyk` target, even if none of its subtargets have been modified.

Reply:
You are right.

> A phony target should not be a prerequisite of a real target file; if it is, its recipe will be run every time make goes to update that file. As long as a phony target is never a prerequisite of a real target, the phony target recipe will be executed only when the phony target is a specified goal (see [Arguments to Specify the Goals](https://www.gnu.org/software/make/manual/html_node/Goals.html#Goals)). 

([Phony Targets](https://www.gnu.org/software/make/manual/html_node/Phony-Targets.html))